core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -386,16 +390,22 @@ private Boolean isFeatureEnabled(@Nonnull ProjectConfig projectConfig,
         FeatureDecision featureDecision = decisionService.getVariationForFeature(featureFlag, userId, copiedAttributes, projectConfig);
         Boolean featureEnabled = false;
         SourceInfo sourceInfo = new RolloutSourceInfo();
+        if (featureDecision.decisionSource != null) {
```

Comment:
in which case, decisionSource can be null.

Reply:
when experiment and variation is null. 
the cases are 
1. When The feature flag is not used in a rollout
2. The rollout with id was not found in the datafile for feature flag